### DISCUSSION TITLE
Nested Updates

### DISCUSSION THREAD

**Discussion by qwacko** [geldata/gel#4226] [General]
Hi, 

I am just getting started with EdgeDB, and really enjoying getting the permissions / validation etc.. so close to (in) the DB. I did have a challenge, for a single user I have a private and public set of data in different object types, and wanted to do an update to to both the private and public in a single transaction, however I am unsure if this is possible. All the nested update documentation is referring to changing the links, but not adjusting the linked data.

Is there any way to update linked data, or just the link? To do what I want to do, would I be best to run multiple update commands in a single transaction, however this would seem to negate some of the benefits of EdgeQL as it would require multiple DB writes rather than one.

Here is an excerpt of a section of my SDL (simplified down). To update the User and UserPrivateDetail types, is there any way to update both of these in a single EdgeQL command (or actually a single typescript query-builder call)?

```edgedb
type User {

        # Properties --------------------------------------------------------------
        required property email -> str {
            readonly := true;
            constraint exclusive;
            constraint regexp(r'[\w\-\.]+@([\w\-]+\.)+[\w\-]{2,4}');
            };        
        required property admin -> bool {
            default := false;
        }

        # Links -------------------------------------------
        required link privateDetail -> UserPrivateDetail {
            readonly := true;
            on source delete delete target;
            constraint exclusive;
        }

 
    }

    # User Private Details and Settings Type ###########################
    ############################################################
    type UserPrivateDetail {

        required property darkMode -> bool {
            default := false;
        }

        # Backlinks ----------------------------------------
        link user := .<privateDetail[is User];

    }
```

**Comment by colinhacks** [comment]
Something like this should work

```
with user := (select HousesAvailability filter ...)
select {
  update_user := (update user set { ... }),
  update_private := (update user.privateDetails set { ... })
}
```

This is using a "free object" to execute two `updates` in a single query - this is a good trick if you ever want to execute a bunch of statements in a single query.

Check out the [Compose queries](https://www.edgedb.com/docs/clients/js/querybuilder#compose-queries) and [Free object](https://www.edgedb.com/docs/clients/js/select#selecting-free-objects) query builder docs for details on how to do this in TypeScript.
